src/wp-includes/rest-api/endpoints/class-wp-rest-widgets-controller.php

```diff
@@ -470,6 +483,16 @@ protected function save_widget( $request ) {
 			$form_data = array();
 		}
 
+		/**
+		 * Fires after a widget is created or updated via the REST API.
+		 *
+		 * @since 5.8.0
+		 * @param WP_Widget       $widget_object Inserted or updated widget object.
+		 * @param WP_REST_Request $request       Request object.
+		 * @param bool            $creating      True when creating a post, false when updating.
+		 */
+		do_action( 'rest_save_widget', $widget_object, $request, $creating );
```

Comment:
I'm having doubts about whether this action is useful at all, actually. There's not a lot that a plugin can do if they hook into it. It might be better if we let plugins filter `$form_data` instead.

For now, I think let's not add this action at all and just have `reset_after_save_widget`.